python-lib/azure_translation_api_client.py

```diff
@@ -0,0 +1,96 @@
+# -*- coding: utf-8 -*-
+"""Module with utility functions to call the Azure Translator API"""
+
+import json
+import logging
+import re
+from typing import AnyStr
+
+import requests
+
+# ==============================================================================
+# CONSTANT DEFINITION
+# ==============================================================================
+
+
+API_EXCEPTIONS = (requests.HTTPError,)
+
+
+# ==============================================================================
+# CLASS AND FUNCTION DEFINITION
+# ==============================================================================
+
+
+class AzureTranslatorClient:
+    """
+    Translation client interfacing with the Azure Translator.
+
+    Args:
+        api_key: Azure Translator API Key
+        location: Azure location, e.g. francecentral
+
+    """
+
+    def __init__(self, api_key, location) -> None:
+        self.api_key = api_key
+        self.location = location
+
+        self.endpoint = "https://api.cognitive.microsofttranslator.com/translate"
+
+    def translate(
+        self,
+        text: AnyStr,
+        target_language: AnyStr,
+        source_language: AnyStr = "",
+    ) -> AnyStr:
+        """
+        Translates text.
+
+        Args:
+            text: Text to be translated
+            target_language: Code of the language into which the text should be translated
+            source_language: Code of the language of the text to be translated
+
+        Returns:
+            response.text: JSON string with the API response.
+
+        Raises:
+            HTTPError: An error occured accessing the API.
+        """
+        response = requests.post(
+            url=self.endpoint,
+            headers={
+                "Ocp-Apim-Subscription-Key": self.api_key,
+                "Ocp-Apim-Subscription-Region": self.location,
+                "Content-type": "application/json",
+            },
+            params={"api-version": "3.0", "from": source_language, "to": target_language},
```

Comment:
Maybe we should keep the api-version as a constant somewhere to keep track of this. In the constant definitions section ?

Reply:
Good idea done